apps/prairielearn/src/server.js

```diff
@@ -2300,22 +2292,26 @@ if (esMain(import.meta) && config.startServer) {
         await opentelemetry.init({
           ...config,
           serviceName: 'prairielearn',
+          // For Sentry to work correctly, it needs to hook into our OpenTelemetry setup.
+          // https://docs.sentry.io/platforms/javascript/guides/node/tracing/instrumentation/opentelemetry/
+          //
+          // However, despite what their documentation claims, only the `SentryContextManager`
+          // is necessary if one isn't using Sentry for tracing. In fact, if `SentrySpanProcessor`
+          // is used, 100% of traces will be sent to Sentry, despite us never having set
+          // `tracesSampleRate` in the Sentry configuration.
+          contextManager: config.sentryDsn ? new Sentry.SentryContextManager() : undefined,
```

Comment:
This (and the downstream change in `@prairielearn/opentelemetry`) are the crucial difference from the last attempt. Apparently this is necessary to ensure context isolation between requests.

Reply:
I'm not following all the details here but I agree that this seems reasonable given their docs and the conversation in the issue you opened on the Sentry SDK.